src/Type/PostObject/Mutation/PostObjectUpdate.php

```diff
@@ -0,0 +1,172 @@
+					$existing_post = get_post( absint( $id_parts['id'] ) );
+
+					/**
+					 * If there's no existing post, throw an exception
+					 */
+					if ( empty( $id_parts['id'] ) || false === $existing_post ) {
+						// translators: the placeholder is the name of the type of post being updated
+						throw new \Exception( sprintf( __( 'No %1$s could be found to update', 'wp-graphql' ), $post_type_object->graphql_single_name ) );
+					}
+
+					if ( $post_type_object->name !== $existing_post->post_type ) {
+						// translators: The first placeholder is an ID and the second placeholder is the name of the post type being edited
+						throw new \Exception( sprintf( __( 'The id %1$d is not of the type "%2$s"', 'wp-graphql' ), $id_parts['id'], $post_type_object->name ) );
+					}
+
+					/**
+					 * Stop now if a user isn't allowed to edit posts
+					 */
+					if ( ! current_user_can( $post_type_object->cap->edit_posts ) ) {
+						// translators: the $post_type_object->graphql_single_name placeholder is the name of the object being mutated
+						throw new \Exception( sprintf( __( 'Sorry, you are not allowed to update a %1$s', 'wp-graphql' ), $post_type_object->graphql_single_name ) );
+					}
+
+					/**
+					 * If the mutation is setting the author to be someone other than the user making the request
+					 * make sure they have permission to edit others posts
+					 */
+					$author_id_parts = ! empty( $input['authorId'] ) ? Relay::fromGlobalId( $input['authorId'] ) : null;
+					if ( ! empty( $author_id_parts[0] ) && get_current_user_id() !== $author_id_parts[0] && ! current_user_can( $post_type_object->cap->edit_others_posts ) ) {
+						// translators: the $post_type_object->graphql_single_name placeholder is the name of the object being mutated
+						throw new \Exception( sprintf( __( 'Sorry, you are not allowed to update %1$s as this user.', 'wp-graphql' ), $post_type_object->graphql_plural_name ) );
+					}
+
+					/**
+					 * @todo: when we add support for making posts sticky, we should check permissions to make sure users can make posts sticky
+					 * @see: https://github.com/WordPress/WordPress/blob/e357195ce303017d517aff944644a7a1232926f7/wp-includes/rest-api/endpoints/class-wp-rest-posts-controller.php#L640-L642
+					 */
+
+					/**
+					 * @todo: when we add support for assigning terms to posts, we should check permissions to make sure they can assign terms
+					 * @see: https://github.com/WordPress/WordPress/blob/e357195ce303017d517aff944644a7a1232926f7/wp-includes/rest-api/endpoints/class-wp-rest-posts-controller.php#L644-L646
+					 */
+
+					/**
+					 * insert the post object and get the ID
+					 */
+					$post_args = PostObjectMutation::prepare_post_object( $input, $post_type_object, $mutation_name );
+					$post_args['ID'] = absint( $id_parts['id'] );
+
+					/**
+					 * Insert the post and retrieve the ID
+					 */
+					$post_id = wp_update_post( $post_args );
```

Comment:
As with `wp_insert_post()` you actually have to pass `true` as the second param. See docs: https://developer.wordpress.org/reference/functions/wp_update_post/